File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        res = 0
        memo = {}
        def dfs(i):
            if i >= len(nums):
                return 0
            if i in memo:
                return memo[i]
            curr = 1
            for j in range(i+1, len(nums)):
                if nums[i] < nums[j]:
                    curr = max(curr, 1 + dfs(j))
            memo[i] = curr
            return curr
        
        for i in range(len(nums)):
            res = max(res, dfs(i))
        return res

File: Data Structures & Algorithms/sliding-window-maximum/submission-1.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        List<Integer> res = new ArrayList<>();
        /*
        Create left pointer at 0, right as iterator = k-1
        Create temp pointer to iterate inside window to get max
        */
        int left = 0;
        for(int right = k-1; right<nums.length; right++){
            int currN = Integer.MIN_VALUE;
            for(int j = left; j<=right; j++){
                currN = Math.max(currN, nums[j]);
            }
            res.add(currN);
            left++;
        }
        int len = res.size();
        int[] trueRes = new int[len];
        for(int i=0; i<res.size(); i++){
            trueRes[i] = res.get(i);
        }
        return trueRes;

    }
}
